sdk/rust/oak/src/lib.rs

```diff
@@ -128,6 +128,32 @@ pub fn wait_on_channels(handles: &[ReadHandle]) -> Result<Vec<ChannelReadStatus>
     }
 }
 
+/// The same as [`wait_on_channels`](#method.wait_on_channels), but also applies the current Node's
+/// downgrade privilege when checking IFC restrictions.
+pub fn wait_on_channels_with_downgrade(
+    handles: &[ReadHandle],
+) -> Result<Vec<ChannelReadStatus>, OakStatus> {
+    let mut space = new_handle_space(handles);
```

Comment:
Not sure whether it is worth it, but you could consider extracting the logic into a util function to avoid the duplication of quite complex logic.